Lib/subprocess.py

```diff
@@ -1307,15 +1307,15 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
                 try:
                     exception_name, hex_errno, err_msg = (
                             errpipe_data.split(b':', 2))
+                    err_msg = err_msg.decode(errors="surrogatepass")
                 except ValueError:
                     exception_name = b'SubprocessError'
                     hex_errno = b'0'
-                    err_msg = (b'Bad exception data from child: ' +
+                    err_msg = ('Bad exception data from child: ' +
                                repr(errpipe_data))
```

Comment:
I suggest to add an "else:" block here and move "err_msg = err_msg.decode(errors="surrogatepass")" there.